connectionpool.go

```diff
@@ -486,7 +483,21 @@ func (pool *hostConnPool) fillingStopped(hadError bool) {
 
 	pool.mu.Lock()
 	pool.filling = false
+	count := len(pool.conns)
+	host := pool.host
+	port := pool.port
 	pool.mu.Unlock()
+
+	// if we errored and the size is now zero, make sure the host is marked as down
+	// see https://github.com/gocql/gocql/issues/1614
+	if gocqlDebug {
+		pool.logger.Printf("gocql: conns of pool after stopped %q: %v\n", host.ConnectAddress(), count)
+	}
+	if err != nil && count == 0 {
+		if pool.session.cfg.ConvictionPolicy.AddFailure(err, host) {
+			pool.session.handleNodeDown(host.ConnectAddress(), port)
```

Comment:
Previously, `handleNodeDown` was called in a new goroutine. Is it safe to call it directly?

Reply:
Yeah, there was a comment about that. Previously it was called in a goroutine to avoid a deadlock but now it's called within `fillingStopped` which is called within an anonymous goroutine at the end of `fill`. So it's still in a goroutine w/r/t `fill`.